docs/decisions/0002-pie-structure.rst

```diff
@@ -0,0 +1,87 @@
+=====================================================
+
+Status
+------
+
+Accepted
+
+What is a Program Intent?
+-------------------------
+
+A program intent records a **learner's** **intent** to complete a **program**. That intent can be positive or negative, certain or uncertain. Each intent also records the **reason** we are storing the intent, such as package purchase or completion of courses within the program.
+
+Intents are never modified. If new information gives us a new belief about the learner's intentions, we write a new intent for that learner and program, leaving the old one in place. Intents have two timestamps to assist with ordering, the standard **created** timestamp and an **effective** timestamp which is the time this intent became true to the system. If a weekly batch job on Saturday were to notice that the user had signed up for a program on Tuesday, created would be Saturday, and effective would be Tuesday.
+
+Model
++++++
+
++--------------------+----------------------------------------+
+|Data                |Note                                    |
++====================+========================================+
+|Program ID          |program UUID is the generic identifier  |
+|                    |                                        |
++--------------------+----------------------------------------+
+|Learner ID          |                                        |
++--------------------+----------------------------------------+
+|Reason              |i.e. ALL_COURSES_COMPLETE or            |
+|                    |BUNDLE_PUCRHASE                         |
++--------------------+----------------------------------------+
+|Certainty           |CERTAIN_YES, CERTAIN_NO or MAYBE        |
+|                    |                                        |
++--------------------+----------------------------------------+
+|Effective Time      |when did the event occur that made this |
+|                    |intent                                  |
++--------------------+----------------------------------------+
+|Bookkeeping fields  |ID, created date, etc                   |
+|for Django          |                                        |
++--------------------+----------------------------------------+
+
+API
+---
+
+Writes
+++++++
+
+
+MVP
+~~~
+
+We need a simple write event call, something like:
+
+POST /programintent (user ID, program ID, type, category, optional effective time if not NOW)
+
+This will write an intent to the store if there is not already an intent for this user/program/type/date. This lets us start writing UIs which ask about intent.
+
+To bootstrap our intents we also need a management command to load intents generated by reports:
+
+LoadAllTheseProgramIntents (intent file)
+
+
+Later Work
+~~~~~~~~~~
+
+As a replacement for the management command we will connect a pipeline from the data warehouse to the PIE service. This will remove the periodic manual update we'll be using during the MVP.
```

Comment:
What is the "periodic manual update" here? Is the assumption we'll periodically run the `loadAllTheseProgramIntents` management command up until this point?

Reply:
that's exactly it - for the MVP we'll have to just carry these in a bucket